src/main/java/org/jitsi/jigasi/transcription/AbstractTranscriptPublisher.java

```diff
@@ -0,0 +1,477 @@
+        }
+        catch (OperationFailedException e)
+        {
+            logger.warn("Failed to send message " + messageString, e);
+        }
+    }
+
+    /**
+     * Save a transcript in a file
+     *
+     * @param transcript the transcript to save
+     */
+    protected void saveTranscriptToFile(String fileName, T transcript)
+    {
+        File logDir = Paths.get(getLogDirPath()).toFile();
+
+        // Try to make the directory
+        if(!logDir.exists())
+        {
+            if(!logDir.mkdir())
+            {
+                logger.warn("Was not able to safe a transcript because" +
+                    " unable to make a directory called " + logDir);
+                return;
+            }
+        }
+
+        // If there is a file with the directory name, we can't make the
+        // directory and thus we cannot save the transcript
+        if(logDir.exists() && !logDir.isDirectory())
+        {
+            logger.warn("Was not able to safe a transcript because" +
+                " there is a file called " + logDir);
+            return;
+        }
+
+        File t = new File(logDir, fileName);
+        try(FileWriter writer = new FileWriter(t))
+        {
+            writer.write(transcript.toString());
+            logger.info("Wrote final transcript to " + t);
+        }
+        catch(IOException e)
+        {
+            logger.warn("Unable to write transcript to file " + t, e);
+        }
+    }
+
+    /**
+     * Get the string representing the path of the directory wherein the
+     * final transcripts should be stored
+     *
+     * @return the path as a String
+     */
+    public String getLogDirPath()
+    {
+        return JigasiBundleActivator.getConfigurationService()
+            .getString(P_NAME_TRANSCRIPT_DIRECTORY,
+                TRANSCRIPT_DIRECTORY_DEFAULT_VALUE);
+    }
+
+    /**
+     * Get the string representing the base URL the server will use to host the
+     * formatted {@link Transcript}
+     *
+     * @return the base URL as a String
+     */
+    protected String getBaseURL()
+    {
+       return JigasiBundleActivator.getConfigurationService()
+           .getString(P_NAME_TRANSCRIPT_BASE_URL,
+               TRANSCRIPT_BASE_URL_DEFAULT_VALUE);
+    }
+
+    /**
+     * Get whether the URL should be advertises when a
+     * {@link TranscriptionGatewaySession} joins a room
+     *
+     * @return true when the URL should be advertises, false otherwise
+     */
+    protected boolean advertiseURL()
+    {
+        return JigasiBundleActivator.getConfigurationService()
+            .getBoolean(P_NAME_ADVERTISE_URL,
+                ADVERTISE_URL_DEFAULT_VALUE);
+    }
+
+    /**
+     * Get a file name for a transcript, which includes the time and some ID
+     * to make it hard to guess
+     *
+     * @return the file name
+     */
+    protected String getHardToGuessFileName()
+    {
+        return "transcript_" + UUID.randomUUID() + "_" + Instant.now();
```

Comment:
Since this is an instance method of a specific publisher, wouldn't it make sense to save the generated value and return it in subsequent calls to getHardToGuessFileName()? Otherwise, I would make it static and rename to "generate" instead of "get" for clarity.